src/ServiceControl/Operations/ErrorIngestion.cs

```diff
@@ -54,9 +54,9 @@ public async Task EnsureStarted()
 
         Task OnCriticalErrorAction(ICriticalErrorContext ctx) => onCriticalError(ctx.Error, ctx.Exception);
 
-        public async Task EnsureStopped()
+        public async Task EnsureStopped(CancellationToken cancellationToken)
         {
-            await startStopSemaphore.WaitAsync().ConfigureAwait(false);
+            await startStopSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Would it make sense to include this into the try / catch finally to make sure we always release?